### PR TITLE
feat(bot): make nitpicks surface-only, remove nitpick gate

### PR DESCRIPTION
**Pull request by rickcedwhat-ai**
## What & why

You found the manual nitpick-checkbox awkward — and digging in, CodeRabbit itself classifies nitpicks as advisory/low-value and posts them in the review *summary*, not as resolvable threads. Gating on them required the checkbox dance + fragile per-comment parsing (which gave us the "🤖 Prompt for AI Agents" wrong-file bug and the thread-count race). This makes nitpicks **surface-only**: visible, never blocking.

## Changes

- **Delete `bot-cr-nitpick-gate.yml`** — no more checkbox gate.
- **`bot-cr-complete` gates on actionable count, not threads.** It now reads CR's `Actionable comments posted: N` line (reliable, in the review body immediately) instead of the GraphQL thread count (which races with GitHub materializing threads — the bug that showed "0 threads"):
  - APPROVED, or **0 actionable + no open threads** → `complete`
  - >0 actionable, or open threads → `unresolved`
  - **A nitpick-only review (0 actionable) now resolves to `complete`** — the core of "don't block on nits"
- **Smoke table simplified to counts** (`N actionable · M nitpicks`) + thread rows. Drops the fragile per-nitpick file/line parse, fixing both display bugs at once.
- **`coderabbit-logic.mjs`:** removed `hasUncheckedNitpicks` + `parseReviewSection`; added `parseActionableCount` + `parseNitpickCount` with unit tests (269 pass, +9).

## How a PR reaches `complete` now

Solely from a fresh CR review that approves or has 0 actionable comments — fix → re-review → done. No checkbox.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Refactor**
  * Updated CodeRabbit review completion detection to use count-based evaluation instead of checkbox-based gating logic.
  * Removed the nitpick gate workflow that previously managed review status transitions and label updates.
  * Simplified review parsing logic to focus on count extraction with improved error handling.

* **Tests**
  * Added tests for new parsing functions covering normal cases, missing input, and edge case scenarios.

### CODE REVIEW

**Review by rickcedwhat — APPROVED** [review]

File: .github/workflows/bot-cr-complete.yml [hunk 1]
```diff
@@ -179,143 +153,85 @@ jobs:
...
             }
 
-            // ── 1C. Build smoke test table ────────────────────────────────────
-            const allRows = [
-              ...inlineThreads,
-              ...outsideDiff.map(c => ({ type: '⚠️ Outside diff', file: c.file, lines: c.lines, title: c.title })),
-              ...nitpicks.map(c => ({ type: '🟡 Nitpick', file: c.file, lines: c.lines, title: c.title })),
-            ];
+            // Block only on actionable comments / open threads. Nitpicks are advisory.
+            const hasBlocking = !isApproved && (actionableCount > 0 || inlineThreads.length > 0);
```
[the remark this answers is not in the document]
Comment (rickcedwhat-ai) [reply .github/workflows/bot-cr-complete.yml:180]: Fixed in cd2096f — `hasBlocking` now includes `reviewState.toLowerCase() === 'changes_requested'`, so a changes-requested review blocks even if the actionable-count parse missed or threads haven't materialized yet.

File: .github/workflows/bot-cr-complete.yml [hunk 2]
```diff
@@ -179,143 +153,85 @@ jobs:
...
 
-            core.info(`CR review ${reviewState} → coderabbit: unresolved`);
+            // ── apply outcome ─────────────────────────────────────────────────
+            if (!hasBlocking) {
+              await swapLabel('coderabbit: complete');
+              await setStatus('success', 'CodeRabbit review passed');
+              const note = nitpickCount > 0
+                ? `> ✅ Review complete · no blocking issues · ${nitpickCount} advisory nitpick(s)`
+                : '> ✅ Review complete · no issues';
+              await writeCRSection(note, nitpickCount > 0);
```
[the remark this answers is not in the document]
Comment (rickcedwhat-ai) [reply .github/workflows/bot-cr-complete.yml:233]: Fixed in cd2096f, but by deletion rather than update. `scripts/coderabbit-check-logic.mjs` was dead code from the old bot — only its own test imported it, no workflow or source referenced it (superseded by `.github/scripts/coderabbit-logic.mjs`). Rather than update `getReviewOutcome` to a policy nothing calls, I removed the module and its test entirely, which clears the contradiction.

**Review by rickcedwhat — APPROVED** [review]

**Review by rickcedwhat — APPROVED** [review]
